beluga/test/beluga/sensor/test_ndt_model.cpp

```diff
@@ -112,31 +114,109 @@ TEST(NDTSensorModelTests, FitPoints) {
         Eigen::Vector2d{0.1, 0.2}, Eigen::Vector2d{0.1, 0.9}, Eigen::Vector2d{0.1, 0.2},
         Eigen::Vector2d{0.1, 0.9}, Eigen::Vector2d{0.1, 0.2}, Eigen::Vector2d{0.1, 0.2},
     };
-    auto cell = detail::fit_points<2>(meas);
+    auto cell = detail::fit_points(meas);
     ASSERT_TRUE(cell.mean.isApprox(Eigen::Vector2d{0.1, 0.433333}, 1e-6));
     ASSERT_FALSE(cell.covariance.isZero());
     ASSERT_GT(cell.covariance(1, 1), cell.covariance(0, 0));
   }
 }
 
-TEST(NDTSensorModelTests, ToCellsNotEnoughPointsInCell) {
-  const double map_res = 0.5;
+TEST(NDTSensorModel2DTests, ToCellsNotEnoughPointsInCell) {
+  constexpr double map_res = 0.5;
   const std::vector map_data{
       Eigen::Vector2d{0.1, 0.2},
       Eigen::Vector2d{0.112, 0.22},
       Eigen::Vector2d{0.15, 0.23},
   };
-  const auto cells = detail::to_cells<2UL>(map_data, map_res);
+  const auto cells = detail::to_cells(map_data, map_res);
+  ASSERT_EQ(cells.size(), 0UL);
+}
+
+TEST(NDTSensorModel3DTests, ToCellsNotEnoughPointsInCell) {
+  constexpr double map_res = 0.5;
+  const std::vector map_data{
+      Eigen::Vector3d{0.1, 0.2, 0.0},
+      Eigen::Vector3d{0.112, 0.22, 0.0},
+      Eigen::Vector3d{0.15, 0.23, 0.0},
+  };
+  const auto cells = detail::to_cells(map_data, map_res);
   ASSERT_EQ(cells.size(), 0UL);
 }
 
-TEST(NDTSensorModelTests, SensorModel) {
-  const double map_res = 0.5;
+TEST(NDTSensorModel3DTests, FitPoints) {
+  {
+    const std::vector meas{
+        Eigen::Vector3d{0.1, 0.2, 0.3}, Eigen::Vector3d{0.1, 0.2, 0.3}, Eigen::Vector3d{0.1, 0.2, 0.3},
+        Eigen::Vector3d{0.1, 0.2, 0.3}, Eigen::Vector3d{0.1, 0.2, 0.3}, Eigen::Vector3d{0.1, 0.2, 0.3},
+    };
+    auto cell = detail::fit_points(meas);
+    ASSERT_TRUE(cell.mean.isApprox(Eigen::Vector3d{0.1, 0.2, 0.3}));
+    // We introduce a minimum variance to avoid numeric errors down the line.
+    ASSERT_FALSE(cell.covariance.isZero());
+  }
+  {
+    const std::vector meas{
+        Eigen::Vector3d{0.1, 0.2, 0.3}, Eigen::Vector3d{0.1, 0.2, 0.3}, Eigen::Vector3d{0.1, 0.2, 0.5},
+        Eigen::Vector3d{0.1, 0.2, 0.3}, Eigen::Vector3d{0.1, 0.9, 0.4}};
+    auto cell = detail::fit_points(meas);
+    ASSERT_TRUE(cell.mean.isApprox(Eigen::Vector3d{0.1, 0.34, 0.36}, 1e-6));
+    ASSERT_FALSE(cell.covariance.isZero());
+    ASSERT_GT(cell.covariance(1, 1), cell.covariance(0, 0));
+    ASSERT_GT(cell.covariance(1, 1), cell.covariance(2, 2));
+    ASSERT_GT(cell.covariance(2, 2), cell.covariance(0, 0));
+  }
+}
+
+TEST(NDTSensorModel3DTests, SensorModel) {
+  constexpr double map_res = 0.5;
```

Comment:
@serraramiro1 nit: consider using Hungarian notation for consistency:
```suggestion
  constexpr double kMapResolution = 0.5;
```
Same elsewhere for other new `constexpr` in this file.

Reply:
DOne